pkg/costattribution/metrics.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package costattribution
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"google.golang.org/protobuf/proto"
+)
+
+// descriptor wraps a prometheus.Desc with additional metadata for metric validation and creation.
+// This wrapper is needed because prometheus.Desc doesn't expose important information about the
+// descriptor itself, such as metric name, label names, and a possible error obtained during the
+// descriptor construction. The first two are accessible via the name and labels fields, while
+// construction error can be obtained by calling validate().
+type descriptor struct {
+	desc   *prometheus.Desc
+	name   string
+	labels []string
+}
+
+// newDescriptor creates a new descriptor with the specified metric name, help text, variable labels, and constant labels.
+func newDescriptor(name string, help string, labels []string, constLabels prometheus.Labels) *descriptor {
+	return &descriptor{
+		desc:   prometheus.NewDesc(name, help, labels, constLabels),
+		name:   name,
+		labels: labels,
+	}
+}
+
+// validate checks whether the underlying prometheus.Desc is valid.
+//
+// A prometheus.Desc may contain an internal error if created with invalid
+// metric or label names. Since this error is unexported, validate detects
+// such cases and returns the underlying error if present. Otherwise, it
+// returns nil.
+func (d *descriptor) validate() error {
+	reg := prometheus.NewRegistry()
+	if err := reg.Register(d); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
As discussed offline, I think this is ok as a workaround, but I also think it's reasonable to have a `Desc.Err()` method or similar, so I'd send a PR to `prometheus/client_golang` to expose it.